scripts/mac_refresh.py

```
@@ -0,0 +1,92 @@
+        cipher = Cipher(algorithms.AES(derived_key), modes.GCM(iv, tag), backend=default_backend())
+        return (cipher.decryptor().update(payload) + cipher.decryptor().finalize()).decode('utf-8', errors='ignore')
+    except: return None
+
+def fetch_xoxc_via_api(d_cookie):
+    """Dynamically fetches a fresh xoxc_token using the valid d_cookie."""
+    try:
+        req = urllib.request.Request(
+            "https://app.slack.com/client",
+            headers={"Cookie": f"d={d_cookie}", "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7)"}
+        )
+        with urllib.request.urlopen(req, timeout=10) as resp:
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)